metadata-ingestion/src/datahub/ingestion/source/vertica/common.py

```diff
@@ -0,0 +1,1573 @@
+        ) and (
+            profile_candidates is None
+            or (profile_candidates is not None and dataset_name in profile_candidates)
+        )
+
+    def loop_profiler_requests_for_projections(
+        self,
+        inspector: Inspector,
+        schema: str,
+        sql_config: SQLAlchemyConfig,
+    ) -> Iterable["GEProfilerRequest"]:
+        from datahub.ingestion.source.ge_data_profiler import GEProfilerRequest
+        tables_seen: Set[str] = set()
+        profile_candidates = None  # Default value if profile candidates not available.
+        if (
+            sql_config.profiling.profile_if_updated_since_days is not None
+            or sql_config.profiling.profile_table_size_limit is not None
+            or sql_config.profiling.profile_table_row_limit is None
+        ):
+            try:
+                threshold_time: Optional[datetime.datetime] = None
+                if sql_config.profiling.profile_if_updated_since_days is not None:
+                    threshold_time = datetime.datetime.now(
+                        datetime.timezone.utc
+                    ) - datetime.timedelta(
+                        sql_config.profiling.profile_if_updated_since_days
+                    )
+                profile_candidates = self.generate_profile_candidates(
+                    inspector, threshold_time, schema
+                )
+            except NotImplementedError:
+                logger.debug("Source does not support generating profile candidates.")
+        for projection in inspector.get_projection_names(schema):
+
+            schema, projection = self.standardize_schema_table_names(
+                schema=schema, entity=projection
+            )
+            dataset_name = self.get_identifier(
+                schema=schema, entity=projection, inspector=inspector
+            )
+
+            if not self.is_dataset_eligible_for_profiling(
+                dataset_name, sql_config, inspector, profile_candidates
+            ):
+                if self.config.profiling.report_dropped_profiles:
+                    self.report.report_dropped(f"profile of {dataset_name}")
+                continue
+            dataset_name = self.normalise_dataset_name(dataset_name)
+            if dataset_name not in tables_seen:
+                tables_seen.add(dataset_name)
+            else:
+                logger.debug(f"{dataset_name} has already been seen, skipping...")
+                continue
+            missing_column_info_warn = self.report.warnings.get(MISSING_COLUMN_INFO)
+            if (
+                missing_column_info_warn is not None
+                and dataset_name in missing_column_info_warn
+            ):
+                continue
+            (partition, custom_sql) = self.generate_partition_profiler_query(
+                schema, projection, self.config.profiling.partition_datetime
+            )
+            if partition is None and self.is_table_partitioned(
```

Comment:
do projections support partitions?

it'd be helpful to add some comments about what projections are and how they're mapped to datahub concepts